android/java/org/chromium/chrome/browser/ntp/IncognitoNewTabPageView.java

```diff
@@ -0,0 +1,229 @@
+    private IncognitoNewTabPageManager mManager;
+    private boolean mFirstShow = true;
+    private NewTabPageScrollView mScrollView;
+
+    private int mSnapshotWidth;
+    private int mSnapshotHeight;
+    private int mSnapshotScrollY;
+
+    private int mWidthDp;
+    private int mHeightDp;
+
+    private static final int BULLETPOINTS_HORIZONTAL_SPACING_DP = 40;
+    private static final int CONTENT_WIDTH_DP = 600;
+    private static final int WIDE_LAYOUT_THRESHOLD_DP = 720;
+
+    /**
+     * Manages the view interaction with the rest of the system.
+     */
+    interface IncognitoNewTabPageManager {
+        /** Loads a page explaining details about incognito mode in the current tab. */
+        void loadIncognitoLearnMore();
+
+        /**
+         * Called when the NTP has completely finished loading (all views will be inflated
+         * and any dependent resources will have been loaded).
+         */
+        void onLoadingComplete();
+    }
+
+    /** Default constructor needed to inflate via XML. */
+    public IncognitoNewTabPageView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+
+        mScrollView = (NewTabPageScrollView) findViewById(R.id.ntp_scrollview);
+        mScrollView.setBackgroundColor(
+                ApiCompatibilityUtils.getColor(getResources(), R.color.ntp_bg_incognito));
+        setContentDescription(
+                getResources().getText(R.string.accessibility_new_incognito_tab_page));
+
+        // FOCUS_BEFORE_DESCENDANTS is needed to support keyboard shortcuts. Otherwise, pressing
+        // any shortcut causes the UrlBar to be focused. See ViewRootImpl.leaveTouchMode().
+        mScrollView.setDescendantFocusability(FOCUS_BEFORE_DESCENDANTS);
+
+        View learnMore = findViewById(R.id.learn_more);
+        learnMore.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mManager.loadIncognitoLearnMore();
+            }
+        });
+
+        mWidthDp = getContext().getResources().getConfiguration().screenWidthDp;
+        mHeightDp = getContext().getResources().getConfiguration().screenHeightDp;
+
+        adjustView();
+    }
+
+    /**
+     * Initialize the incognito New Tab Page.
+     * @param manager The manager that handles external dependencies of the view.
+     */
+    void initialize(IncognitoNewTabPageManager manager) {
+        mManager = manager;
+    }
+
+    /** @return The IncognitoNewTabPageManager associated with this IncognitoNewTabPageView. */
+    protected IncognitoNewTabPageManager getManager() {
+        return mManager;
+    }
+
+    /**
+     * @see org.chromium.chrome.browser.compositor.layouts.content.
+     *         InvalidationAwareThumbnailProvider#shouldCaptureThumbnail()
+     */
+    boolean shouldCaptureThumbnail() {
```

Comment:
Not needed?

Reply:
removed

Reply:
I found out that we need it, reverted back, not sure how did it manage to build when I removed it.